packages/patterns/src/types.js

```diff
@@ -524,17 +529,23 @@ export {};
 /** @typedef {(...args: any[]) => any} Method */
 
 /**
- * @template {Record<string | symbol, MethodGuard>} [T=Record<string | symbol, MethodGuard>]
+ * @template {Record<PropertyKey, MethodGuard>} [T=Record<PropertyKey, MethodGuard>]
  * @typedef {{
  *   interfaceName: string,
- *   methodGuards: { [K in keyof T]: K extends symbol ? never : T[K] },
- *   symbolMethodGuards?: CopyMap<(keyof T) & symbol, MethodGuard>,
+ *   methodGuards:
+ *     { [K in Exclude<keyof T, symbol>]: T[K] } &
+ *     Partial<{ [K in Extract<keyof T, symbol>]: never }>,
+ *   symbolMethodGuards?:
+ *     CopyMap<
+ *       Extract<keyof T, symbol>,
+ *       T[Extract<keyof T, symbol>]
+ *     >,
```

Comment:
Huh, interesting. I'm surprised that something like `T[keyof T]` works, but indeed it does.